### PR TITLE
Feature/pz 25 reusable button component

### PR DESCRIPTION
**Pull request by sarojghalan**
# Merge Request : feature/pz 25 reusable button component

## Summary of change
In this pr i have created a styled component of button, where its types and interface are also created for type-checking, for more i have created a global theme provider to provide theme across the project, and a global theme which contains overall used color of the project.

## Mention

Review: @subhashishshah @dipeshkumarsah98 

Fixes: #task_issue_id

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [X] New feature (non-breaking change which adds functionality)
- [ ] **Breaking change** (fix or feature that would cause existing functionality to not work as expected)
- [X] Front End (UI) changes
- [ ] Changes in `documentation`

## How did you test

- [ ] Tested the functionality of A
- [ ] Tested the functionality of B

## Checklist

- [X] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [X] My code follows the style guidelines of this project
- [ ] I have made corresponding changes to the documentation
- [X] My changes generate no new warnings
- [ ] I have added required tests that prove my fix is effective or that my feature works
- [ ] Both New and Existing unit tests pass locally with my changes

### CODE REVIEW

File: components/PZButton/style.ts [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+    color: white;
+    padding: 6px 26px;
+    font-size: 14px;
+    text-transform: uppercase;
+    transition: all 0.25s ease-in-out;
+    border: 1px solid black;
+    &:hover {
+      background-color: white !important ;
+      color: black !important;
+      border: 1px solid black !important;
```
Comment (dipeshkumarsah98) [inline components/PZButton/style.ts:15]: while hovering cursor should be pointer

**Review by dipeshkumarsah98 — CHANGES_REQUESTED** [review]
Nice Work :+1: .The global theme is not passed to the app as a result compilation error is encountered. 
And formatting is not setup as well. use pretter to format the code. I'm getting error when i open you edited files. Resolve the mentioned issues and ask for re-review. Thank you
image

image

File: components/PZButton/style.ts (hunk 1 above)
Comment (sarojghalan) [reply components/PZButton/style.ts:15]: sorry my bad.

File: app/layout.tsx [hunk 2]
```diff
@@ -14,7 +15,9 @@ export default function RootLayout({ children }: { children: React.ReactNode })
   return (
     <html lang='en'>
       <StyledComponentsRegistry>
-        <body className={inter.className}>{children}</body>
+        <body className={inter.className}>
```
Comment (subhashishjungshah) [inline app/layout.tsx:18]: please remove this inter.classname. I think we aren't using this font.

File: components/PZButton/index.tsx [hunk 3]
```diff
@@ -0,0 +1,11 @@
+import React from 'react'
+import { StyledPZButton } from './style'
+import { IPZButtonProps } from './type'
+
+const PZButton = ({ children, className, variant, onClick }: IPZButtonProps) => (
+  <StyledPZButton className={`${className} ${variant}`} onClick={onClick}>
```
Comment (subhashishjungshah) [inline components/PZButton/index.tsx:6]: why aren't we using classname package?

File: components/PZButton/style.ts [hunk 4]
```diff
@@ -0,0 +1,34 @@
+import styled from 'styled-components'
+
+export const StyledPZButton = styled.button`
+  &.primary__btn {
```
Comment (subhashishjungshah) [inline components/PZButton/style.ts:4]: okay there are alot of repeated css. for example, we need to group the same css in one place. Only the extra required for primary__btn should be written inside &.primary__btn and &.square__btn :100:

File: package.json [hunk 5]
```diff
@@ -26,6 +26,7 @@
     "@types/react": "18.2.14",
     "@types/react-dom": "18.2.6",
     "autoprefixer": "^10.4.14",
+    "classnames": "^2.3.2",
```
Comment (subhashishjungshah) [inline package.json:29]: please use this package if you have installed.

File: app/layout.tsx [hunk 6]
```diff
@@ -1,5 +1,6 @@
 import { Inter } from 'next/font/google'
 import StyledComponentsRegistry from './registry'
+import { PZThemeProvider } from './theme-provider'
 import type { Metadata } from 'next'
 import './globals.css'
```
Comment (subhashishjungshah) [inline app/layout.tsx:5]: what is inside ./globals.css?

**Review by subhashishjungshah — COMMENTED** [review]
@sarojghalan please resolve these comments!

File: components/PZButton/style.ts (hunk 1 above)
Comment (dipeshkumarsah98) [reply components/PZButton/style.ts]: when you solve the comment please click on a resolve conversation button to resolve the comment.

File: components/PZButton/style.ts (hunk 4 above)
Comment (dipeshkumarsah98) [inline components/PZButton/style.ts:4]: variants name is passes a props in a component so it should be better to follow camelCase rule. like primaryButton.

File: components/PZButton/style.ts [hunk 7]
```diff
@@ -0,0 +1,34 @@
...
+    transition: all 0.25s ease-in-out;
+    border: 1px solid black;
+    &:hover {
+      background-color: ${({ theme }) => theme.color.white__color};
+      color: black;
+      border: 1px solid ${({ theme }) => theme.color.black__color};
+      cursor: pointer;
+    }
+  }
+  &.square__btn {
```
Comment (dipeshkumarsah98) [inline components/PZButton/style.ts:19]: use addittioanl props name as size where you pass the sizes such as 'sm', 'lg', 'md', 'xl'. And the variants should be common like primary, secondary

**Review by sarojghalan — COMMENTED** [review]
conversation resolved

**Pull request merged**